### PR TITLE
(IMAGES-416) Fix Vagrant ssh behaviour

### PR DESCRIPTION
**Pull request by jcoconnor**
Revise the usernames used for the build process so that
"vagrant/vagrant" is used throughout including the
boxstarter-initialisation.

Make sure the Administrator account profile is also created, with an
empty directory for the Administrator Home dir on ssh.

Finally run sysprep to clear the profiles and pick up the default user
settings provided by puppet.

### CODE REVIEW

**Comment by jcoconnor** [comment]
@jpogran @glennsarti & @McdonaldSeanp I have added you as reviewers on this work.
I had to jump through quite a few hoops to get ssh working as per what the modules team were looking for. Still some issues, but this may be with hour beaker is used on Windows.

/cc @DavidS @tphoney

**Comment by DavidS** [comment]
@jcoconnor can you drop me a build of this for testing somewhere? I'll be in the office tomorrow.

**Comment by jcoconnor** [comment]
@davids - Yes going to put it up on Google Drive for the moment - its a lot faster than ```int-resources```

File: scripts/windows/start-boxstarter.ps1 [hunk 1]
```diff
@@ -22,8 +26,8 @@ Remove-Item -Path "$($Env:USERPROFILE)\Desktop\Boxstarter Shell.lnk" -Confirm:$f
 Remove-Item -Path "$($Env:APPDATA)\Microsoft\Windows\Start Menu\Programs\Boxstarter" -Recurse -Confirm:$false -Force -ErrorAction SilentlyContinue | Out-Null
 
 # Use Admin Plaintext password for this phase of configuration
-$secpasswd = ConvertTo-SecureString "PackerAdmin" -AsPlainText -Force
-$cred = New-Object System.Management.Automation.PSCredential ("Administrator", $secpasswd)
+$secpasswd = ConvertTo-SecureString "$LoginPassword" -AsPlainText -Force
```
Comment (glennsarti) [inline scripts/windows/start-boxstarter.ps1]: 👍

File: templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            <UserData>
+                <!-- Product Key from http://technet.microsoft.com/en-us/library/jj612867.aspx -->
+                <ProductKey>
+                    <!-- Do not uncomment the Key element if you are using trial ISOs -->
+                    <!-- KMS Setup keys  https://technet.microsoft.com/en-us/library/jj612867.aspx -->
+                    <Key>D2N9P-3P6X9-2R39C-7RTCD-MDVJX</Key>
+                    <WillShowUI>OnError</WillShowUI>
+                </ProductKey>
+                <AcceptEula>true</AcceptEula>
+                <FullName>Puppet Labs</FullName>
```
Comment (glennsarti) [inline templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml]: Is this a PuppetLabs provided Vagrant box?

File: templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                <HideEULAPage>true</HideEULAPage>
+                <HideLocalAccountScreen>true</HideLocalAccountScreen>
+                <HideOEMRegistrationScreen>true</HideOEMRegistrationScreen>
+                <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
+                <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
+                <NetworkLocation>Work</NetworkLocation>
+                <ProtectYourPC>1</ProtectYourPC>
+            </OOBE>
+            <RegisteredOwner />
+            <ProductKey>D2N9P-3P6X9-2R39C-7RTCD-MDVJX</ProductKey>
```
Comment (glennsarti) [inline templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml]: Strange...We already set PID earlier.

File: templates/windows-2012r2/x86_64.virtualbox.vagrant.cygwin.json [hunk 4]
```diff
@@ -109,18 +109,31 @@
...
+    {
+      "type": "powershell",
+      "inline": [
         "C:\\Packer\\Init\\vagrant-arm-host.ps1"
       ]
     }
   ],
     "post-processors": [{
     "type"                : "vagrant",
+    "compression_level"   : 6,
```
Comment (glennsarti) [inline templates/windows-2012r2/x86_64.virtualbox.vagrant.cygwin.json]: Curious as to what this does

File: templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml (hunk 2 above)
Comment (jcoconnor) [inline templates/windows-2012r2/files/virtualbox/post-clone.autounattend.xml]: Yes - for internal use only - its not going onto our public vagrant box site.
Its the same intro as for the vmpooler boxes.

**Comment by jcoconnor** [comment]
@Davids & @tphoney You should be able to download the .box file from: https://drive.google.com/open?id=0B2_YrJBsZpg5Mks0aWxaVi1FQm8

**Comment by ScottGarman** [comment]
:+1:

File: scripts/windows/init/vagrant-arm-host.ps1 [hunk 5]
```diff
@@ -10,31 +10,43 @@ $AdministratorHome = "$ENV:CYGWINDIR\home\Administrator"
...
 Write-Host "Setting SSH Host Configuration"
-$qa_root_passwd = Get-Content "$ENV:CYGWINDOWNLOADS\qapasswd"
-& $CygWinShell --login -c `'ssh-host-config --yes --privileged --user cyg_server --pwd $qa_root_passwd`'
+& $CygWinShell --login -c `'ssh-host-config --yes --privileged --user cyg_server --pwd vagrant`'
 
 # Generate ssh keys.
 Write-Host "Generate SSH Keys"
-& $CygWinShell --login -c `'rm -rf /home/Administrator/.ssh/id_rsa*`'
-& $CygWinShell --login -c `'ssh-keygen -t rsa -N `"`" -f /home/Administrator/.ssh/id_rsa`'
+& $CygWinShell --login -c `'rm -rf /home/vagrant/.ssh/id_rsa*`'
```
Comment (Iristyle) [inline scripts/windows/init/vagrant-arm-host.ps1]: Seems like the user should get pulled up to a variable within this script

**Pull request merged**